### PR TITLE
[Enhancement][Bugfix] Update Java content

### PR DESCRIPTION
**Pull request by redhatrises**
- Fix various path issues
- Update STIG reference file
- Add OCIL tags and content
- Update bash scripts, XCCDF, and OVAL to use /etc/.java instead of ${deployment.java.home}
  as /etc/.java has precendence over ${deployment.java.home}
- Remove OS template files
- Update CPE to use generic JAVA tags

### CODE REVIEW

File: Java/input/application/java.xml [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+</Rule>
+
+<Rule id="java_jre_deployment_config_configured" severity="medium">
+<title>The Java Configuration File Is Configured Correctly</title>
+<description>
+To ensure that the Java <tt>/etc/.java/deployment/deployment.config</tt> file is
+configured correctly, add or modify
+<tt>deployment.system.config</tt> to equal 
+<tt>file:///etc/.java/deployment/deployment.properties</tt>. Also add or modify
+<tt>deployment.system.config.mandatory</tt> to equal <tt>false</tt>.
```
Comment (shawndwells) [inline Java/input/application/java.xml]: Should this be broken into two XCCDF rules, since two configuration actions are required?

**Comment by redhatrises** [comment]
@shawndwells I can break it out into two different rules if so desired. Just an FYI, the stig rule is not broken into two different XCCDF rules (Not sure if that makes or should make a difference or not on whether it gets broken into two separate XCCDF rules).

**Comment by redhatrises** [comment]
Created a sub-group and broke out the XCCDF for `deployment.config` into 2 separate rules.

**Comment by iankko** [comment]
Makes sense && looks good to me AFAICT from testing. There's small issue when having Java content enabled && issuing ```make / make dist```:
```
oscap ds sds-compose output/ssg-java-xccdf-1.2.xml output/ssg-java-ds.xml
File '/root/scap-security-guide/Java/output/ssg-java-ds.xml' line 295: Element '{http://oval.mitre.org/XMLSchema/oval-definitions-5}variables': Missing child element(s). Expected is one of ( {http://oval.mitre.org/XMLSchema/oval-definitions-5}variable, {http://oval.mitre.org/XMLSchema/oval-definitions-5}external_variable, {http://oval.mitre.org/XMLSchema/oval-definitions-5}constant_variable, {http://oval.mitre.org/XMLSchema/oval-definitions-5}local_variable ).
Invalid SCAP Source Datastream content(1.2) in /root/scap-security-guide/Java/output/ssg-java-ds.xml.
make: *** [content] Error 1
```

But that isn't problem of this PR (rather it's problem of common build process not considering the scenario when OVAL file in question doesn't contain ```<variables/>``` at all:
```
# pwd
/root/scap-security-guide/Java/input/checks
# grep -rHn "_variable" * | wc -l
0
```

Note: The issue wasn't visible before because there have been some variable definitions present in the macros, that got removed in this PR. I will submit a PR fixing the build process shortly.

ACK for merging this.

**Comment by redhatrises** [comment]
@iankko thanks for reviewing! I can submit a PR for that since I introduced that bug.

**Pull request merged**